### PR TITLE
Multiple fixes

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox run -m lint`?
- [ ] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

**Review by mr-cal — APPROVED** [review]
Looks good, just need to run isort

File: tests/unit/commands/test_validation_sets.py [hunk 1]
```diff
@@ -345,7 +345,9 @@ def test_edit_validation_sets_with_errors_not_amended(
     assert confirm_mock.mock_calls == [call("Do you wish to amend the validation set?")]
 
 
-def test_edit_yaml_error_retry(mocker, tmp_path):
+def test_edit_yaml_error_retry(mocker, tmp_path, monkeypatch):
+    monkeypatch.setenv("EDITOR", "vi")
```
Comment (lengau) [inline tests/unit/commands/test_validation_sets.py:349]: Good catch! Should we change this to a nonexistent command somewhere (perhaps in an auto-use fixture somewhere) to catch any time we make this assumption?

File: tests/unit/services/test_package.py [hunk 2]
```diff
@@ -383,7 +382,7 @@ def test_update_project_parse_info(
     )
 
     parse_info = {"my-part": ["file.metadata.xml"]}
-    package = Package(
+    package = services.Package(
```
Comment (lengau) [inline tests/unit/services/test_package.py:385]: ❤️

**Review by lengau — APPROVED** [review]

File: tests/unit/commands/test_validation_sets.py (hunk 1 above)
Comment (sergiusens) [reply tests/unit/commands/test_validation_sets.py:349]: will do

**Pull request merged**